wrapper/src/main/java/software/amazon/jdbc/util/WrapperUtils.java

```diff
@@ -251,7 +260,9 @@ public static <T, E extends Exception> T executeWithPlugins(
       }
 
     } finally {
-      pluginManager.unlock();
+      if (lock.isHeldByCurrentThread()) {
```

Comment:
I wonder if the unlock should happen after calling closeContext. Can you check with @sergiyvamz